### PR TITLE
Add ConcreteMoistureTimeIntegrationMassLumped

### LINKED ISSUE #327: Add version of ConcreteMoistureTimeIntegration with mass lumping to minimize oscillations
## Reason
Mass lumping is an accepted procedure to minimize oscillatory behavior in the presence of sharp concentration fronts (see https://mooseframework.inl.gov/modules/porous_flow/mass_lumping.html). This behavior is sometimes observed in the moisture solution for concrete, and it is expected that it will benefit from mass lumping.

## Design
Add a new version of this kernel that derives from MassLumpedTimeDerivative.

## Impact
This will only add new features. We may want to deprecate the non-lumped version of this at some point.

### PR DESCRIPTION
**Pull request by bwspenc**
closes #327

### CODE REVIEW

**Comment by jiangwen84** [comment]
Looks good. Only one minor comment.

**Comment by bwspenc** [comment]
> Looks good. Only one minor comment.

Thanks, @jiangwen84 ! I can't see your comment for some reason. If you can point me to that, that would be great.

**Review by jiangwen84 — CHANGES_REQUESTED** [review]

**Review by jiangwen84 — DISMISSED** [review]
Looks good to me.

**Comment by jiangwen84** [comment]
@bwspenc  I do not think I have the right to merge it. It looks good to me. You can have someone to merge it.

File: include/kernels/ConcreteMoistureTimeIntegrationMassLumped.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/****************************************************************/
+
+#pragma once
+
+// MOOSE includes
+#include "MassLumpedTimeDerivative.h"
+
+// Forward Declarations
+class ConcreteMoistureTimeIntegrationMassLumped;
+
```
Comment (dschwen) [inline include/kernels/ConcreteMoistureTimeIntegrationMassLumped.h:22]:
```suggestion
```

not needed

File: include/kernels/ConcreteMoistureTimeIntegrationMassLumped.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ConcreteMoistureTimeIntegrationMassLumped(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual() override;
+  virtual Real computeQpJacobian() override;
+
+  /// Material property for moisture capacity (dW/dH). Note that only
+  /// a subset of the moisture diffusivity models provide this.
+  /// The other models assume that this is 1.
+  const MaterialProperty<Real> * _moisture_capacity; //  dW/dH
```
Comment (dschwen) [inline include/kernels/ConcreteMoistureTimeIntegrationMassLumped.h:40]:
```suggestion
  const MaterialProperty<Real> * const  _moisture_capacity; //  dW/dH
```

File: src/kernels/ConcreteMoistureTimeIntegrationMassLumped.C [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      "Time derivative term with mass lumping for moisture transport in concrete.");
+  return params;
+}
+
+ConcreteMoistureTimeIntegrationMassLumped::ConcreteMoistureTimeIntegrationMassLumped(
+    const InputParameters & parameters)
+  : MassLumpedTimeDerivative(parameters), _moisture_capacity(nullptr)
+{
+  if (hasMaterialProperty<Real>("moisture_capacity"))
+    _moisture_capacity = &getMaterialProperty<Real>("moisture_capacity");
```
Comment (dschwen) [inline src/kernels/ConcreteMoistureTimeIntegrationMassLumped.C:33]:
```suggestion
  : MassLumpedTimeDerivative(parameters), 
    _moisture_capacity(hasMaterialProperty<Real>("moisture_capacity") ? &getMaterialProperty<Real>("moisture_capacity") : nullptr)
{
```

just a suggestion!

**Review by dschwen — APPROVED** [review]

**Pull request merged**